openwisp_controller/config/whois/tasks.py

```
@@ -0,0 +1,114 @@
+            )
+        logger.error(f"WHOIS lookup failed. Details: {exc}")
+        return super().on_failure(exc, task_id, args, kwargs, einfo)
+
+
+# device_pk is used when task fails to report for which device failure occurred
+@shared_task(
+    bind=True,
+    base=WHOISCeleryRetryTask,
+    **app_settings.API_TASK_RETRY_OPTIONS,
+)
+def fetch_whois_details(self, device_pk, initial_ip_address):
```

Ruff (0.14.14)

Unused function argument: `self`

(ARG001)